### PR TITLE
Triggering Jenkins and Plan a Release

### PR DESCRIPTION
**Pull request by lilyhdo**
Please review

### CODE REVIEW

File: planning-release.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Planning a release on XebiaLabs
```
Comment (tiennguyenhoan) [inline planning-release.md:1]: As user, I think there are should have a introduce in README to provide some high-level information about this Repo:

1.  Add an agenda in README which help user know which section they are going to read. 
2. Provide some brief information about "XebiaLabs-Appium-ReactNative", e.g What will this Repo show to the end user.

File: planning-release.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+In XL Release, create a new release.
```
Comment (tiennguyenhoan) [inline planning-release.md]: As user, I don't know what "XL Release" mean for the first time reading this, and the sentence really confuse me. Please make it clearer 

Suggestion: In the top navigation bar, looking for the tab "Release", ...  click on "New Release" button ....

File: planning-release.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Planning a release on XebiaLabs
+
+In XL Release, create a new release. 
+
```
Comment (tiennguyenhoan) [inline planning-release.md]: As discussed, please make it clearer. I really confuse me when I reading this.

File: planning-release.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+![new-release](assets/new-release.png)
+
+Name your release. Notice that under 'Release details', you can specify the date which you want the release to start, among other things. 
+
+![create-new-release](assets/create-new-release.png)
+
+In a release, you can plan several phases for the project. This will allow you to manage, control, and optimize your process easily.
```
Comment (tiennguyenhoan) [inline planning-release.md]: Where is the "release" ?
Where can I plan the phases ?
How can "plan several phases" help me manage, control, and optimize better ? 

Suggestion: Trying to break this part in to sub sections. 
1. Showing that is the release user have created from the above part
2. Add a part for introducing "Add Phase" and giving some images or information. 
3. Add some brief reason for manage, control, and optimize better

File: planning-release.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+In a release, you can plan several phases for the project. This will allow you to manage, control, and optimize your process easily. 
+
+![phases](assets/phases.png)
+
+Each phase is comprised of many task, such as testing, making sure prerequisites are fulfilled, or building. 
+
+To add a new task, click on 'Add task' under the desired phase. 
+
+![add-task](assets/add-task.png)
```
Comment (tiennguyenhoan) [inline planning-release.md]: This image can use for introduce new Phase

File: planning-release.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+Name your release. Notice that under 'Release details', you can specify the date which you want the release to start, among other things. 
+
+![create-new-release](assets/create-new-release.png)
+
+In a release, you can plan several phases for the project. This will allow you to manage, control, and optimize your process easily. 
+
+![phases](assets/phases.png)
+
+Each phase is comprised of many task, such as testing, making sure prerequisites are fulfilled, or building.
```
Comment (tiennguyenhoan) [inline planning-release.md]: 1. You can explain this line with the image above, like: "Let assume the whole process of release will have 4 Phrase (Build, Test, Deploy, and Announce). In each phase, ...." 

2. I don't understand, why do I need to add a task for every phrase ?

File: planning-release.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+![phases](assets/phases.png)
+
+Each phase is comprised of many task, such as testing, making sure prerequisites are fulfilled, or building. 
+
+To add a new task, click on 'Add task' under the desired phase. 
+
+![add-task](assets/add-task.png)
+
+In the dropdown menu, you can specify the type of task.
```
Comment (tiennguyenhoan) [inline planning-release.md]: Image for this one ?

File: planning-release.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+Each phase is comprised of many task, such as testing, making sure prerequisites are fulfilled, or building. 
+
+To add a new task, click on 'Add task' under the desired phase. 
+
+![add-task](assets/add-task.png)
+
+In the dropdown menu, you can specify the type of task. 
+
+With Xebia XL, you can plan all the steps of your pipeline and link it to multiple other platforms such as GitHub or Jenkins.
```
Comment (tiennguyenhoan) [inline planning-release.md]: This statement seem off-topic ?

File: triggering-jenkins.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
+# Triggering a Jenkins build as a task on XebiaLabs
+
+This guide will demonstrate how to set a XebiaLabs task to trigger a Jenkins build. 
+
+## Prerequisites
+
+* Jenkins is installed
+
+* Have an account with XebiaLabs
+
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md:10]: The structure is not clear, is the "1.1 Create a new Jenkins job" a part of Prerequisites
I think we should add some thing like "## Guidline"

File: triggering-jenkins.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+#### 1.1 Create a new Jenkins job
+
+On the left side of the site, click on 'New Item' to create a new project.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: "left side of the site" ?

why don't we use "left sidebar" ?

File: triggering-jenkins.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+```
+#!/bin/bash
+[ ! -d "samples" ] && git clone https://github.com/kobiton/samples.git || echo "Samples already exist"
+cd samples/javascript
+npm install
+npm run android-app-test
+```
+
+In your automation test, make sure to include your own Kobiton username and API key as well as the desired device and platform you wish to test with, which can be found on the Kobiton website.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: This sententces isn't relevant to the topic. I think this is a note so make it separate with the content
Suggestion: 
> Note: In your automation test, .....

By the way, this line seem hard to understand, please simplify it.

File: triggering-jenkins.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+This guide is executing the NodeJS test script from the Kobiton samples, so we will add the NodeJS Plugin on Jenkins to execute NodeJS script as a build step. To do this, go to 'Manage Jenkins' -> 'Manage Plugins' -> Available. Search for NodeJS and install it.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: I completely get lost by this part, I don't understand the relevant between this part and the above parts ? Is it a separate part ? 

I would like to suggest the structure: 
```
# Triggering a Jenkins build as a task on XebiaLabs
## Prerequisites
....
##  Jenkins configuration 
### 1. Integrate NodeJs Plugin
....

### 2. Create a new Jenkins job
....

## XebiaLabs Configuration
### 1. Install the Jenkins plugin
...
### 2. Update Jenkins plugin settings
...

... so on
```

By the ways, NodeJS Plugin must be installed or just an optional for the sample above ?

File: triggering-jenkins.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+![manage-plugins](assets/manage-plugins.png)
+
+If you are using a different language, you may try to find a suitable plugin for your project if necessary.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: what do you mean by different language ?

File: triggering-jenkins.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+Under build environment, check the box that says 'Provide Node & npm bin/ folder to PATH'.
+
+![node-path](assets/node-path.png)
+
+Save your configuration settings. Try 'Build Now' and then check Kobiton cloud devices to see if a test session was created. If successful, you can move onto triggering a Jenkins build from a XebiaLabs task.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: "you can move onto triggering a Jenkins build from a XebiaLabs task" Is it related to the `Create a new Jenkins job` ???

File: triggering-jenkins.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+
+If you are using a different language, you may try to find a suitable plugin for your project if necessary.
+
+Under build environment, check the box that says 'Provide Node & npm bin/ folder to PATH'.
+
+![node-path](assets/node-path.png)
+
+Save your configuration settings. Try 'Build Now' and then check Kobiton cloud devices to see if a test session was created. If successful, you can move onto triggering a Jenkins build from a XebiaLabs task. 
+
+#### 2.1 Install the Jenkins plugin
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: Install the Jenkins plugin in XebiaLabs

File: triggering-jenkins.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+
+On XebiaLabs, click on 'Plugins' in the navigation bar at the top of the screen. In the search bar, type in 'Jenkins' and install it.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: "the navigation bar" -> The **top** navigation bar

File: triggering-jenkins.md [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+
+Enter in a title for the configuration, as well as the Jenkins url where the server can be reached. 
+
+Under 'Authentication', enter in your Jenkins username and password. 
+
+![server-settings](assets/server-settings.png)
+
+#### 2.3 Set Jenkins as a task
+
+If you do not already have a release, create a new one.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: Why don't we add the link to planning-release.md ? 

Suggestion: 
```
If you have not created a release yet, follow [this link](./planning-release.md) for guiding. 
```

File: triggering-jenkins.md [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+
+Click on 'Add task' under the desired phase. 
+
+![add-task](assets/add-task.png)
+
+Set a name for the new task. Click on the dropdown menu and select 'Jenkins' and 'Build'. Then click the 'Add' button to finish creating the task. 
+
+![build](assets/build.png)
+
+> You can also run an automation test script directly as a task in XebiaLabs, but there may be issues for Jython and Groovy scripts.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: I think this sentences distract the topic

**Review by tiennguyenhoan — CHANGES_REQUESTED** [review]
@lilyhdo I have added some comments, please check it.

File: triggering-jenkins.md (hunk 12 above)
Comment (lilyhdo) [reply triggering-jenkins.md]: For this example that's executing a NodeJS script, the plugin is necessary. I will make it more clear.

File: README.md [hunk 19]
```diff
@@ -1,2 +1,11 @@
 # XebiaLabs-Appium-ReactNative
 Guidance on integrating Kobiton service into the mobile app build pipeline: XebiaLabs, Appium and ReactNative.
```
Comment (tiennguyenhoan) [inline README.md]: I think you can remove this line, it seem duplicate meaning with "The guidelines in this repo will..."

File: planning-release.md [hunk 20]
```diff
@@ -0,0 +1,39 @@
+# Planning a release on XebiaLabs
+
+With XebiaLabs's XL Release, you can plan, automate, and optimize your pipeline visually in a neat, step-by-step diagram which is referred to as a "release".
```
Comment (tiennguyenhoan) [inline planning-release.md]: Is this part necessary to have in this file ?
I think we miss-understanding from the previous comment.

``` 
2. Provide some brief information about "XebiaLabs-Appium-ReactNative", e.g What will this Repo show to the end user.
```

I mean you should add some information in README.md.

File: planning-release.md [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+
+To add a new task, click on 'Add task' under the desired phase. 
+
+![add-task](assets/add-task.png)
+
+In the dropdown menu, you can specify the type of task. Tasks will execute specific steps in the release process. In this example, the task we add will run an automation script. 
+
+![create-task](assets/create-task.png)
+
+With Xebia XL, you can plan all the steps of your pipeline through phases and tasks.
```
Comment (tiennguyenhoan) [inline planning-release.md]: I don't what relevant between this line and above parts ?

File: triggering-jenkins.md [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+
+### 1. Integrate NodeJS Plugin
+
+This guide will execute the NodeJS test script from the Kobiton samples for this Jenkins build. To run a NodeJS script, installing the NodeJS Plugin on Jenkins is necessary. 
+
+Add the NodeJS Plugin on Jenkins to execute NodeJS script as a build step. To do this, go to 'Manage Jenkins' -> 'Manage Plugins' -> Available. Search for NodeJS and install it.
+
+![manage-plugins](assets/manage-plugins.png)
+
+If you are using a different programming language for your automation script, you may try to find a suitable plugin for your project if necessary.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: "> Note: If you are using...."

File: README.md (hunk 19 above)
Comment (tiennguyenhoan) [inline README.md]: You can remove this line, I think the meaning of it is duplicating with "The guidelines in this repo will show you how to plan a release with the XebiaLabs service XL release as well as how to trigger a Jenkins build from a task on XebiaLabs."

File: planning-release.md (hunk 20 above)
Comment (tiennguyenhoan) [inline planning-release.md]: I think we are miss-understanding from the previous comment. 

```
2. Provide some brief information about "XebiaLabs-Appium-ReactNative", e.g What will this Repo show to the end user.
```

I mean that you should provide some brief information about "XebiaLabs-Appium-ReactNative" in README.md and you have done it in the newest commit

File: planning-release.md (hunk 21 above)
Comment (tiennguyenhoan) [inline planning-release.md]: I don't know what relevant between this line and the above parts ?

File: triggering-jenkins.md [hunk 23]
```diff
@@ -0,0 +1,99 @@
...
+
+On the left sidebar, click on 'New Item' to create a new project. 
+
+![new-item](assets/new-item.png)
+
+Enter in a project name and select the project type 'Freestyle project'. 
+
+![new-project](assets/new-project.png)
+
+In the configuration settings, go the 'Build' section. Click on 'Add build step' and select 'Execute shell'.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: As user, how can I find the "configuration setting" for the created project.

File: triggering-jenkins.md [hunk 24]
```diff
@@ -0,0 +1,99 @@
...
+
+![server-settings](assets/server-settings.png)
+
+### 3. Set Jenkins as a task
+
+If you have not created a release yet, follow [this link](./planning-release.md) for guidance on how.
+
+![new-release](assets/new-release.png)
+
+Click on 'Add task' under the desired phase.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: I feel lost in this part.

File: triggering-jenkins.md [hunk 25]
```diff
@@ -0,0 +1,99 @@
...
+
+![build](assets/build.png)
+
+Click on the task to open up the settings. Enter in the Server name, which was created earlier. Enter your Jenkins username and password, as well as the name of the job you want to trigger in Jenkins. 
+
+![task-settings](assets/task-settings.png)
+
+Exit out of the settings and start your release. You have successfully created a task that will trigger a Jenkins build!
+
+Check the log output in your task to see if the build was successful.
```
Comment (tiennguyenhoan) [inline triggering-jenkins.md]: "You might need to check the logs output....

**Review by tiennguyenhoan — CHANGES_REQUESTED** [review]

**Comment by tiennguyenhoan** [comment]
@duylam LGTM

File: README.md [hunk 26]
```diff
@@ -1,2 +1,10 @@
 # XebiaLabs-Appium-ReactNative
```
Comment (duylam) [inline README.md:1]: Please help remove `.DS_Store` file

File: README.md [hunk 27]
```diff
@@ -1,2 +1,10 @@
...
-Guidance on integrating Kobiton service into the mobile app build pipeline: XebiaLabs, Appium and ReactNative.
+
+XebiaLabs is a service that develops Continuous Delivery and DevOps software to provide companies greater control over their software delivery process. In XL Release, you can plan, automate, and optimize your pipeline.
+
+The guidelines in this repo will show you how to plan a release with the XebiaLabs service XL release as well as how to trigger a Jenkins build from a task on XebiaLabs.
```
Comment (duylam) [inline README.md]: How about `This guideline will show you to include executing tests on Kobiton devices in your release plan in XebiaLabs  XL Release`

File: README.md [hunk 28]
```diff
@@ -1,2 +1,10 @@
...
+
+## Contents
+ 1. First guild will demonstrate [how to plan a release on XL Release](planning-release.md).
+
+ 2. The second guide will demonstrate [how to trigger a Jenkins build from a task on XL Release](triggering-jenkins.md).
```
Comment (duylam) [inline README.md]: how to trigger a Jenkins build from a task on XL Release --> how to integrate executing tests on Kobiton devices through triggering a Jenkins job

**Review by duylam — COMMENTED** [review]
minor comments

File: triggering-jenkins.md [hunk 29]
```diff
@@ -0,0 +1,219 @@
...
+
+![manage-plugins](assets/2-manageplugins.png)
+
+If you are using a different language, you may try to find a suitable plugin for your project if necessary.
+
+Under build environment, check the box that says 'Provide Node & npm bin/ folder to PATH'.
+
+![build-environment](assets/2-build-environment.png)
+
+**5. Trigger Jenkins project by pushing on GitHub**
```
Comment (duylam) [inline triggering-jenkins.md]: Remove step 5

File: triggering-jenkins.md [hunk 30]
```diff
@@ -0,0 +1,219 @@
...
+
+![build-execute-shell](assets/2-build-execute-shell.png)
+
+Enter this in the command. Enter your own Kobiton  username and API key as well as the desired device and platform you wish to test with. The below example will execute the android-web-test script sample from Kobiton. 
+```
+#!/bin/bash
+[ ! -d "samples" ] && git clone https://github.com/kobiton/samples.git || echo "Samples already exist"
+cd samples/javascript
+npm install
+npm run android-web-test
```
Comment (duylam) [inline triggering-jenkins.md]: `KOBITON_API_KEY=... KOBITON_USERNAME=... npm run android-web-test`

File: triggering-jenkins.md [hunk 31]
```diff
@@ -0,0 +1,219 @@
...
+
+```
+curl -X GET https://api.kobiton.com/v1/sessions \
+  -H 'Authorization: Basic dGVzdHVzZXI6MTIzZWQtMTIzZmFjLTkxMzdkY2E='
+  -H 'Accept: application/json'
+```
+
+Below are necessary Kobiton Rest API endpoints that you may need.
+
+- Get session info
```
Comment (duylam) [inline triggering-jenkins.md]: `[Get session info](https://api.kobiton.com/docs/#get-a-session)`

File: triggering-jenkins.md [hunk 32]
```diff
@@ -0,0 +1,219 @@
...
+  -H 'Authorization: Basic dGVzdHVzZXI6MTIzZWQtMTIzZmFjLTkxMzdkY2E='
+  -H 'Accept: application/json'
+```
+
+With this line of code, you can print the session information to the console. 
+```javascript
+console.log(sessionCapabilities)
+```
+
+- Get session commands
```
Comment (duylam) [inline triggering-jenkins.md]: [Get session commands](https://api.kobiton.com/docs/#get-session-commands)

**Review by duylam — COMMENTED** [review]
Please help check comments

**Pull request merged**